pkg/kyma/cmd/test/run/cmd.go

```diff
@@ -0,0 +1,155 @@
+package run
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	oct "github.com/kyma-incubator/octopus/pkg/apis/testing/v1alpha1"
+	"github.com/kyma-project/cli/pkg/kyma/cmd/test"
+	"github.com/kyma-project/cli/pkg/kyma/cmd/test/client"
+	"github.com/kyma-project/cli/pkg/kyma/core"
+	"github.com/spf13/cobra"
+)
+
+type command struct {
+	opts *options
+	core.Command
+}
+
+func NewCmd(o *options) *cobra.Command {
+	cmd := command{
+		Command: core.Command{Options: o.Options},
+		opts:    o,
+	}
+
+	cobraCmd := &cobra.Command{
+		Use:     "run",
+		Short:   "Run tests on a running Kyma cluster",
+		Long:    `Run tests on a running Kyma cluster`,
+		RunE:    func(_ *cobra.Command, _ []string) error { return cmd.Run() },
+		Aliases: []string{"r"},
+	}
+
+	cobraCmd.Flags().StringVarP(&o.Name, "name", "n", "", "Name for the new test suite")
+	cobraCmd.Flags().StringVarP(&o.Tests, "tests", "s", "", "Test names to execute. Example: --tests=cluster-users-test,test-api-controller-acceptance. Overlaps with --all")
+	cobraCmd.Flags().BoolVarP(&o.RunAll, "all", "a", false, "Run all test cases in the test suite. Overlaps with --tests")
+	cobraCmd.Flags().BoolVarP(&o.Wait, "wait", "w", false, "Wait for test execution to finish")
+	cobraCmd.Flags().IntVarP(&o.Timeout, "timeout", "t", 120, "Timeout for test execution (in seconds)")
+	return cobraCmd
+}
+
+func (cmd *command) Run() error {
+	var err error
+
+	cli, err := client.NewTestRESTClient(10 * time.Second)
+	if err != nil {
+		return fmt.Errorf("unable to create test REST client. E: %s", err)
+	}
+
+	var testSuiteName string
+	if len(cmd.opts.Name) > 0 {
+		testSuiteName = cmd.opts.Name
+	} else {
+		rand.Seed(time.Now().UTC().UnixNano())
+		rnd := rand.Int31()
+		testSuiteName = fmt.Sprintf("test-%d", rnd)
+	}
+
+	tNotExists, err := verifyIfTestNotExists(testSuiteName, cli)
+	if err != nil {
+		return err
+	}
+	if !tNotExists {
+		return fmt.Errorf("test suite '%s' already exists\n", testSuiteName)
+	}
+
+	var testDefToApply []oct.TestDefinition
+
+	clusterTestDefs, err := cli.ListTestDefinitions()
+	if err != nil {
+		return fmt.Errorf("unable to get list of test definitions. E: %s",
+			err.Error())
+	}
+
+	if cmd.opts.RunAll {
+		if cmd.opts.Tests != "" {
+			return fmt.Errorf("-all flag is incompatible with --tests")
+		}
+	} else {
+		if cmd.opts.Tests == "" {
+			return fmt.Errorf("neither --tests nor --all flags provided")
```

Comment:
as mentioned earlier, default should be "all"

Reply:
Fixed